File: solutions/3688-bitwise-or-of-even-numbers-in-an-array/solution.py
# Problem: 3688. Bitwise OR of Even Numbers in an Array
# Difficulty: Easy
# Language: Python
# Runtime: 3 ms
# Memory: 18.1 MB
# Link: https://leetcode.com/problems/bitwise-or-of-even-numbers-in-an-array/

class Solution:
    def evenNumberBitwiseORs(self, nums: List[int]) -> int:
        res=0
        for num in nums:
            if num%2==0:
                res=res|num
        return res

File: solutions/3688-bitwise-or-of-even-numbers-in-an-array/solution.cpp
// Problem: 3688. Bitwise OR of Even Numbers in an Array
// Difficulty: Easy
// Language: C++
// Runtime: 0 ms
// Memory: 24.5 MB
// Link: https://leetcode.com/problems/bitwise-or-of-even-numbers-in-an-array/

class Solution {
public:
    int evenNumberBitwiseORs(vector<int>& nums) {
        int res=0;
        for (int i=0;i<nums.size();i++)
        {
            if (nums[i]%2==0)
            {
                res=res|nums[i];
            }
        }
        return res;
    }
};
